jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -133,9 +134,39 @@ public void assignGroupToRole(final String serverTemplateId, final String contai
     }
 
     @Override
-    public void assignGroupAndUserToRole(final String serverTemplateId, final String containerId, final String caseId, final String roleName, final String user, final String group) {
-        client.assignGroupToRole(containerId, caseId, roleName, group);
-        client.assignUserToRole(containerId, caseId, roleName, user);
+    public void saveAssignment(final String serverTemplateId, final String containerId, final String caseId,
+                               final String roleName, final String users, final String groups) {
+        CaseRoleAssignmentSummary ras = getCaseInstance(serverTemplateId, containerId, caseId).getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(null);
+        List<String> usersList = getListFromString(users);
```

Comment:
Removing and adding all assignments seems unnecessary, could we just add/remove what was actually changed?